doc/guides/west/why.rst

```diff
@@ -6,15 +6,87 @@ History and Motivation
 West was added to the Zephyr project to fulfill two fundamental requirements:
 
 * The ability to work with multiple Git repositories
-* The ability to provide a user-friendly command-line interface to the Zephyr
-  build system and debug mechanisms
-
-Additionally, it was desired that west be easily extensible by
-downstream users.
+* The ability to provide an extensible and user-friendly command-line interface
+  for basic Zephyr workflows
 
 During the development of west, a set of :ref:`west-design-constraints` were
 identified to avoid the common pitfalls of tools of this kind.
 
+Requirements
+************
+
+Although the motivation behind splitting the Zephyr codebase into multiple
+repositories is outside of the scope of this page, the fundamental requirements,
+along with a clear justification of the choice not to use existing tools and
+instead develop a new one, do belong here.
+
+The basic requirements are:
+
+* **R1**: Keep externally maintained code in separately maintained repositories
+  outside of the main zephyr repository, without requiring users to manually
+  clone each of the external repositories
+* **R2**: Provide a tool that both Zephyr users and distributors can make use of
+  to benefit from and extend
+* **R3**: Allow users and downstream distributions to override or remove
+  repositories without having to make changes to the zephyr repository
+* **R4**: Support both continuous tracking and commit-based (bisectable) project
+  updating
+
+Topologies supported
+********************
+
```

Comment:
One small, `tree`-like ASCII art for each example would really help. See example in https://github.com/zephyrproject-rtos/west/issues/237

Reply:
Sure, I'll see what I can do. As you can see in the diff to `repo-tool.rst` in this pull request, I am updating the `tree` output there, which I think is the most important place to have it.

Reply:
> One small, tree-like ASCII art for each example would really help. See example in zephyrproject-rtos/west#237

Done, please let me know what you think.